core/server/src/main/java/alluxio/worker/file/UnderFileSystemManager.java

```diff
@@ -133,6 +139,7 @@ private InputStreamAgent(long sessionId, long agentId, AlluxioURI ufsUri, Config
       mUri = ufsUri.toString();
       mConfiguration = conf;
       UnderFileSystem ufs = UnderFileSystem.get(mUri, mConfiguration);
+      ufs.connectFromWorker(conf, NetworkAddressUtils.getLocalHostName(conf));
```

Comment:
I just noticed that Master uses the following to get hostname:

`ufs.connectFromMaster(conf, NetworkAddressUtils.getConnectHost(ServiceType.MASTER_RPC, conf));`

It's a bit different from simply `getLocalHostname`, I think it's safer to use `getConnectHost` as well?

```
  public static String getConnectHost(ServiceType service, Configuration conf) {
    if (conf.containsKey(service.mHostNameKey)) {
      String connectHost = conf.get(service.mHostNameKey);
      if (!connectHost.isEmpty() && !connectHost.equals(WILDCARD_ADDRESS)) {
        return connectHost;
      }
    }
    if (conf.containsKey(service.mBindHostKey)) {
      String bindHost = conf.get(service.mBindHostKey);
      if (!bindHost.isEmpty() && !bindHost.equals(WILDCARD_ADDRESS)) {
        return bindHost;
      }
    }
    return getLocalHostName(conf);
  }
```